docs/user-guide/binned-data/filtering.ipynb

```diff
@@ -24,8 +24,8 @@
     "\n",
     "### Extract events matching parameter value\n",
     "\n",
-    "Use [label-based indexing on the `bins` property](../../generated/classes/scipp.Bins.rst#scipp.Bins.__getitem__).\n",
-    "This works similar to regular [label-based indexing](../slicing.rst#Label-based-indexing) but operates on the unordered bin contents.\n",
+    "Use [label-based indexing on the bins property](../../generated/classes/scipp.Bins.rst#scipp.Bins).\n",
```

Comment:
Can't we link to `__getitem__`? Or refer to this more clearly?

Reply:
`__getitem__` is not documented. We would have to tell sphinx to autodocument all dunder methods. Not sure if we want to do that

Reply:
Hmm, I actually had to do something like this for ScippNexus, since `__getitem__` is quite complex and requires docs. See https://github.com/scipp/scippnexus/blob/main/docs/_templates/scipp-class-template.rst.

I suppose Scipp is not dissimilar?

Reply:
Added `__getitem__` to the docs. But the links still don't work. It seems like sphinx has a problem with underscores in the name.

Also note that for some classes, the method is documented like
```
__getitem__(*args, **kwargs)

    Overloaded function.

        __getitem__(self: scipp._scipp.core.DataArray, arg0: int) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: slice) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: scipp._scipp.core.Variable) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: Tuple[str, scipp._scipp.core.Variable]) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: Tuple[str, int]) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: Tuple[str, slice]) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: ellipsis) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: List[int]) -> scipp._scipp.core.DataArray

        __getitem__(self: scipp._scipp.core.DataArray, arg0: Tuple[str, List[int]]) -> scipp._scipp.core.DataArray
```
and it doesn't show up in the table of contents style list of methods but only the detailed listing.

Reply:
I see you changed the template, can we now change this links back to point to `__getitem__`?

Reply:
Hooray... well let us worry about this another time, no need to hold up the link fix.